sherpa/env_var_test.go

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2018-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sherpa_test
+
+import (
+	"os"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"github.com/sclevine/spec"
+
+	"github.com/paketo-buildpacks/libpak/sherpa"
+)
+
+func testEnvVar(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect = NewWithT(t).Expect
+	)
+
+	context("AppendToEnvVar", func() {
+
+		context("No Existing", func() {
+
+			it("append one", func() {
+				Expect(sherpa.AppendToEnvVar("TEST_KEY", "|", "test-value-2")).
+					To(Equal("test-value-2"))
+			})
+
+			it("appends multiple", func() {
+				Expect(sherpa.AppendToEnvVar("TEST_KEY", "|", "test-value-2", "test-value-3")).
+					To(Equal("test-value-2|test-value-3"))
+			})
+		})
+
+		context("With Existing", func() {
+			it.Before(func() {
+				Expect(os.Setenv("TEST_KEY", "test-value-1")).To(Succeed())
```

Comment:
The tests are so fast already that is not a big win, but worth noting, that `spec` is capable of running tests and parallel. We can't take advantage of that now because we use the same global state in multiple tests. If we faked the env in our tests or used randomized key name (would need to be done in several additional place) we could start running the tests in parallel.